src/View/TemplatePaths.php

```diff
@@ -241,17 +241,11 @@ public function resolveTemplateFileForControllerAndActionAndFormat($controller,
 		$action = ucfirst($action);
 		$identifier = $controller . '/' . $action . '.' . $format;
 		if (!array_key_exists($identifier, self::$resolvedFiles['templates'])) {
-			foreach ($this->getTemplateRootPaths() as $templateRootPath) {
-				$candidate = $this->ensureAbsolutePath($templateRootPath . $action . '.' . $format);
-				if (file_exists($candidate)) {
-					return self::$resolvedFiles['templates'][$identifier] = $candidate;
-				}
-			}
-			foreach ($this->getTemplateRootPaths() as $templateRootPath) {
-				$candidate = $this->ensureAbsolutePath($templateRootPath . $identifier);
-				if (file_exists($candidate)) {
-					return self::$resolvedFiles['templates'][$identifier] = $candidate;
-				}
+			$templateRootPaths = $this->getTemplateRootPaths();
+			try {
+				return self::$resolvedFiles['templates'][$identifier] = $this->resolveFileInPaths($templateRootPaths, $controller . '/' . $action, $format);
+			} catch (InvalidTemplateResourceException $error) {
+				return self::$resolvedFiles['templates'][$identifier] = $this->resolveFileInPaths($templateRootPaths, $action, $format);
```

Comment:
not sure, but don't we potentially then throw an exception here which was not thrown before?